src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -1,25 +1,24 @@
 import { connect } from 'react-redux';
-import URLSearchParams from 'url-search-params';
 
-import { isLoggedIn } from 'platform/user/selectors';
 import { checkKeepAlive } from 'platform/user/authentication/actions';
 import {
   ssoeInbound,
   hasCheckedKeepAlive,
 } from 'platform/user/authentication/selectors';
+import { hasSession } from 'platform/user/profile/utilities';
 import { checkAutoSession } from 'platform/utilities/sso';
 import { removeLoginAttempted } from 'platform/utilities/sso/loginAttempted';
 
 function AutoSSO(props) {
   const {
     useInboundSSOe,
+    authenticatedWithSSOe,
     hasCalledKeepAlive,
-    userLoggedIn,
     application = null,
     to = null,
   } = props;
 
-  if (userLoggedIn) {
+  if (hasSession()) {
```

Comment:
@hinzed1127 we set the keepAlive call after the async function checkAutoSession returns.  What if we switched that flag on before we call the function?

Reply:
I'm not sure I understand your question here. This is the vets-api session flag. Are you referring to that, or the hasSessionSSO flag?

Reply:
sorry the comments in a bad spot, I'm talking about lines 26 to 28.  The `props.checkKeepAlive()` call is made after `checkAutoSession` returns, what if we called that BEFORE we invoked `checkAutoSession`?

Reply:
It shouldn't matter now, because there's only one other prop initialized from state, but it would make sense to safeguard against changes that include more reudx state properties. previously, `userInboundSSOe` and `userLoggedIn` would both cause a re-render. moving `checkKeepAlive` would help avoid multiple calls to the keepalive